### PR TITLE
Significantly speed up gulp build --css-only

### LINKED ISSUE #9640: All common `gulp` commands should be able to run from a clean state
Currently it's necessary to run `gulp build --css-only` before `gulp dist` to avoid errors like:
```sh
Error compiling ./src/service-worker/kill.js
WARNING - Failed to load module "../../../build/amp-ad-0.1.css.js"
WARNING - Failed to load module "../../../build/amp-form-0.1.css.js"
0 error(s), 2 warning(s)
```
The issue is missing extension-specific CSS*, however `gulp build` also does a bunch of work in addition to that. We can speed up our builds by making sure that `compileCss()` and `buildExtensions()` (CSS only) are run before compiling entry points.

*Another option is to figure out why extension-specific CSS is necessary for building srcs like service-worker. First guess is the quirk that Closure compiler (compile.js) checks the existence of the list of source files to build.

/cc @rsimha-amp @erwinmombay

### PR DESCRIPTION
**Pull request by rsimha-amp**
Prior to this PR, `gulp build --css-only` would take ~ 1 minute to run, and end up building a bunch of js files in addition to compiling css.

After this PR, `gulp build --css-only` takes ~ 0.5 seconds to run (a 100x speed up), and the only output files in `build/` after running it are .css, and .css.js files.

Resulting speeds on my workstation:
`gulp build --css-only`: ~500ms
`gulp build`: ~1m 45s
`gulp dist --fortesting`: ~3m 30s

Fixes #9640

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -182,6 +182,18 @@ function declareExtensionVersionAlias(name, version, lastestVersion, hasCss) {
...
 }
 
+
+/**
+ * Logs a build step to the console, unless we are on Travis, where we don't.
+ * @param {string} stepName Name of an action, like 'Bundling' or 'Compiling'
+ * @param {string} targetName Name of the target, like a filename or path
+ */
+function logBuildStep(stepName, targetName) {
+  if(!process.env.TRAVIS) {
```
Comment (choumx) [inline gulpfile.js]: Space after `if`.

File: gulpfile.js [hunk 2]
```diff
@@ -182,6 +182,18 @@ function declareExtensionVersionAlias(name, version, lastestVersion, hasCss) {
   }
 }
 
+
+/**
+ * Logs a build step to the console, unless we are on Travis, where we don't.
```
Comment (choumx) [inline gulpfile.js]: Nit: "where we don't" is superfluous.

Comment (rsimha-amp) [inline gulpfile.js]: Done.

File: gulpfile.js (hunk 1 above)
Comment (rsimha-amp) [inline gulpfile.js]: Done.

**Comment by rsimha-amp** [comment]
We don't log steps on Travis because the logs will become more verbose than we want. Currently, build is only a small part of the Travis PR logs.

**Comment by rsimha-amp** [comment]
@choumx, I've removed the spurious compileCss from compile. Feel free to take another look.

**Pull request merged**